### DISCUSSION TITLE
Importazione alunni non possibile

### DISCUSSION THREAD

**Discussion by sempliciniluigi89** [iisgiua/giuaschool#539] [Richieste di aiuto]
Buongiorno ,non appena importa il file csv fittizio di alunni per la demo ,ricevo il seguente messaggio:I campi di intestazione del file non corrispondono a quanto previsto.

potrebbe cortesemente sistemare ciò che è sbagliato in modo da inviarmi il file corretto ?Grazie mille e mi scusi il disturbo .
[importazione alunni demo - IMPORTA_ALUNNI_GENITORI (4) (1) (1) (3).csv](https://github.com/iisgiua/giuaschool/files/11871050/importazione.alunni.demo.-.IMPORTA_ALUNNI_GENITORI.4.1.1.3.csv)

**Comment by trinko** [comment]
Era sbagliato il nome del campo "dataNascita".
Allego la versione corretta.

[importazione.alunni.demo.-.IMPORTA_ALUNNI_GENITORI.4.1.1.3.csv](https://github.com/iisgiua/giuaschool/files/11875003/importazione.alunni.demo.-.IMPORTA_ALUNNI_GENITORI.4.1.1.3.csv)

**Reply by sempliciniluigi89** [reply]
Ho provato ad importare la versione da voi corretta e  mi restituisce sempre il seguente messaggio " I campi di intestazione del file non corrispondono a quanto previsto"

**Reply by sempliciniluigi89** [reply]
Mi servirebbe gentilmente il file corretto ,se è possibile .Grazie mille per la pazienza infinita

**Reply by giadaschiavone** [reply]
Ciao Luigi. Hai commesso un errore che ha portato allo sbaglio. Nel file, ha inserito ", sia nelle righe, sia nelle colonne (es. ''nome'', ''Matteo''.) quando dovrebbe essere senza " (es. nome, Matteo). Inoltre, sono presenti molti errori, come la scrittura delle colonne (es. cognome), che non dovrebbe essere scritto tutto in minuscolo, ma con la lettera iniziale in maiuscolo (es. Cognome). Potrebbero esserci anche eventuali errori nei dati che ha inserito (per esempio, mancano i genitori o dei componenti mancati) e non ha indicato i dati e/o colonne obbligatori, come l'username, la scelta delle attività alla Religione Cattolica o all'attività alternativa, la classe frequentata ed eventuali. Credo che la ricostruzione del file ci vorrà molto tempo, di conseguenza, dovresti correggere tu questi errori. Buona giornata, spero di esserti stata d'aiuto.

**Comment by giadaschiavone** [comment]
Ciao Luigi. Hai commesso un errore che ha portato allo sbaglio. Nel file, ha inserito _"_, sia nelle righe, sia nelle colonne (es. _''nome'', ''Matteo''._) quando dovrebbe essere senza _"_ (es. _nome_, _Matteo_). Inoltre, sono presenti molti errori, come la scrittura delle colonne (es. _cognome_), che non dovrebbe essere scritto tutto in minuscolo, ma con la lettera iniziale in maiuscolo (es. _Cognome_). Potrebbero esserci anche eventuali errori nei dati che ha inserito (per esempio, mancano i genitori o dei componenti mancati) e non ha indicato i dati e/o colonne obbligatori, come l'username, la scelta delle attività alla Religione Cattolica o all'attività alternativa, la classe frequentata ed eventuali. Credo che la ricostruzione del file ci vorrà molto tempo, di conseguenza, dovresti correggere tu questi errori. Buona giornata, spero di esserti stata d'aiuto.

**Comment by sempliciniluigi89** [comment]
Antonello mi potrebbe aiutare?

**Reply by sempliciniluigi89** [reply]
@trinko Antonello mi potrebbe aiutare?

**Comment by sempliciniluigi89** [comment]
Buongiorno, nessuno mi sa aiutare a sistemare il file csv  d'importazione degli alunni?

**Comment by trinko** [comment]
Ho testato il file seguente e mi risulta funzionare senza errori:

[importazione-alunni-demo.csv](https://github.com/iisgiua/giuaschool/files/11898669/importazione-alunni-demo.csv)


Unico problema può essere un errore di timeout del server (la procedura richiede troppo tempo e il server la interrompe). In tal caso devi suddividere il file in diversi file più piccoli e eseguirli uno alla volta.
